release.sh

```diff
@@ -5,6 +5,31 @@ cd $DIR
 
 set -e
 
+git remote update
+
+remote=$(git rev-parse "@{u}") # "@{u}" gets the current upstream branch
+local=$(git rev-parse @) # '@' gets the current local branch 
+
+# check if local commit syncs with remote 
+if [ "$remote" != "$local" ]
+    then
+        printf "%s" "Error: local commit does not match remote. Exiting release script." $'\n'
+        exit 1
+fi 
+
+# remove any excess brackets, space/tab characters and 'origin' branch, then sort the tags
+remoteTags=$(git ls-remote --tags origin | grep -v '{}' | sort | tr -d [[:blank:]])
+localTags=$(git show-ref --tags | grep -v '{}' | grep -v 'origin'| sort | tr -d [[:blank:]])
+
+# check if sorted local tags match remote tags
+if ! diff -q <(printf "%s" "$remoteTags") <(printf "%s" "$localTags") >/dev/null 2>&1
```

Comment:
Can you change the above to:

```
git_remote_tags() {
    git ls-remote ...
}
git_local_tags() {
    git show-ref ...
}
if !diff -q <(git_remote_tags) <(git_local_tags) &> /dev/null; then
    ...
fi
```

I think it makes it easier to read and then there doesn't have to be an explanation for why `printf`. I also prefer `&> /dev/null` because it means the same thing as what you have, but it's more explicit in my opinion.
```